plugins/modules/jenkins_plugin.py

```diff
@@ -315,11 +324,14 @@
 import json
 import os
 import tempfile
+import time
+import base64
+from collections import OrderedDict
 
 from ansible.module_utils.basic import AnsibleModule, to_bytes
 from ansible.module_utils.six.moves import http_cookiejar as cookiejar
 from ansible.module_utils.six.moves.urllib.parse import urlencode
-from ansible.module_utils.urls import fetch_url, url_argument_spec
+from ansible.module_utils.urls import fetch_url, url_argument_spec, open_url
```

Comment:
Note that changing from `fetch_url` to `open_url` can easily introducing breaking changes, since `fetch_url` does quite a few things implicitly (and what it does can also depend on the ansible-core version). Switching while still using the `ansible.builtin.url` docs fragment and the `url_argument_spec()` function is generally a very bad idea, since you then need to handle all changes to ansible-core in this collection as well. I would really stick with `fetch_url`. (You then have to modify `module.params` to pass over the right credentials, though...)